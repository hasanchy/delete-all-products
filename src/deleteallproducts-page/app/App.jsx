import { ErrorBoundary } from "react-error-boundary";
import { useEffect } from "react";
import { useDispatch } from "react-redux";
import { __ } from '@wordpress/i18n';
import Header from "../components/header/Header";
import MenuTabs from "../components/menu-tabs/MenuTabs";
import { fetchProductsStat, fetchSettings } from "../services/apiService";

const App = () => {

	const dispatch = useDispatch();

	useEffect(() => {
		dispatch(fetchProductsStat());
		dispatch(fetchSettings());
	}, [])

	return (
		<div className="wrap">
			<Header />
			<ErrorBoundary fallback={<div>{ __( 'Something went wrong', 'delete-all-products' ) }</div>}>
				<MenuTabs/>
			</ErrorBoundary>
		</div>
	)
}

export default App;