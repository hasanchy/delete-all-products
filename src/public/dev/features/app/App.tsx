import Tabs from '../tabs/Tabs';
import Notifications from '../notifications/Notifications';
import { ErrorBoundary } from "react-error-boundary";

const App = (): JSX.Element => {

	return (
		<div className="wrap">
			<h1 style={{ fontSize: '30px' }}><span style={{ color: '#674399' }}>Delete All Products</span></h1>
			<ErrorBoundary fallback={<div>Something went wrong</div>}>
				<Tabs />
				<Notifications />
			</ErrorBoundary>
		</div>
	)
}

export default App;