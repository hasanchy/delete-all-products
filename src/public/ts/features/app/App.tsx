import Tabs from '../tabs/Tabs';
import { ErrorBoundary } from "react-error-boundary";

const App = (): JSX.Element => {

	return (
		<div className="wrap">
			<h1 style={{ fontSize: '30px' }}><span style={{ color: '#674399' }}>Delete WooCommerce Products</span></h1>
			<ErrorBoundary fallback={<div>Something went wrong</div>}>
				<Tabs />
			</ErrorBoundary>
		</div>
	)
}

export default App;