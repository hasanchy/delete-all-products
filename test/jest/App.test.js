import renderer from 'react-test-renderer';
import App from '../../src/public/ts/features/app/App';

it('changes the class when hovered', () => {
	const component = renderer.create(
		<App />,
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});