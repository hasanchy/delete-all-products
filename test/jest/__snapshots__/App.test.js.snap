// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<div
  className="wrap"
>
  <h1
    style={
      {
        "fontSize": "30px",
      }
    }
  >
    <span
      style={
        {
          "color": "#674399",
        }
      }
    >
      Delete WooCommerce Products
    </span>
  </h1>
  <div>
    Something went wrong
  </div>
</div>
`;
